### PR TITLE
Add revision data for WorkloadEndpoint

### PR DESCRIPTION
**Pull request by robbrockbank**
Add to ObjectMetadata the revision info and plumb through for WEP
Add tests

### CODE REVIEW

File: lib/client/workloadendpoint_e2e_test.go [hunk 1]
```diff
@@ -184,8 +164,8 @@ var _ = testutils.E2eDatastoreDescribe("WorkloadEndpoint tests", testutils.Datas
 					"prod": "no",
 				}},
 			api.WorkloadEndpointMetadata{
-				Name:         "ep1/with_foo",
```
Comment (gunjan5) [inline lib/client/workloadendpoint_e2e_test.go]: curious why you removed the `/` from the name?

Comment (robbrockbank) [inline lib/client/workloadendpoint_e2e_test.go]: oh dammit - I actually copied this file across from my refactor branch.

I should really add that back in.  I will.

File: lib/client/workloadendpoint.go [hunk 2]
```diff
@@ -63,7 +64,16 @@ func (w *workloadEndpoints) Apply(a *api.WorkloadEndpoint) (*api.WorkloadEndpoin
...
+	// Validate the supplied Metadata.
+	if err := validator.Validate(metadata); err != nil {
+		return err
+	}
+
+	k, err := w.convertMetadataToKey(metadata)
+	if err != nil {
+		return err
+	}
+	return w.c.Backend.Delete(&model.KVPair{Key: k, Revision: metadata.Revision})
```
Comment (gunjan5) [inline lib/client/workloadendpoint.go]: why did you change the if/else chain here?Couldnt we just add Revision in the call to backend? 
```go
// Untyped get interface for deleting a single API object.  This is called from the typed
// interface.
func (c *Client) delete(metadata unversioned.ResourceMetadata, helper conversionHelper) error {
	// Validate the supplied Metadata.
	if err := validator.Validate(metadata); err != nil {
		return err
	}

	if k, err := helper.convertMetadataToKey(metadata); err != nil {
		return err
	} else if err := c.Backend.Delete(&model.KVPair{Key: k, Revision: metadata.Revision}); err != nil {
		return err
	} else {
		return nil
	}
}
```

Comment (gunjan5) [inline lib/client/workloadendpoint.go]: ah ok nvm you wanted to keep this limited to WEP

Comment (gunjan5) [inline lib/client/workloadendpoint.go]: mind sticking a comment so others don't trip when they modify `delete` in client.go and dont see it being reflected for WEP?

Comment (robbrockbank) [inline lib/client/workloadendpoint.go]: I could add it generically I suppose.  It's not difficult to do.

Comment (robbrockbank) [inline lib/client/workloadendpoint.go]: Actually - I think I will..  It will effectively be limited to WEP because no other resource type will fill in the Revision.

Comment (gunjan5) [inline lib/client/workloadendpoint.go]: naw, lets keep the scope of this PR limited to WEP until your refactor is in :)

Comment (robbrockbank) [inline lib/client/workloadendpoint.go]: it's actually a more compact code change i think.

Comment (gunjan5) [inline lib/client/workloadendpoint.go]: im mostly worried about other things using this. There might be things we havent' considered that might break, for example something "working" even if it was not doing a proper CaS might break with this. Not sure

Comment (robbrockbank) [inline lib/client/workloadendpoint.go]: If someone explicitly fills in the Revision then yes a Delete operation might fail - but in a way that is business as usual.

File: lib/client/client.go [hunk 3]
```diff
@@ -273,9 +273,12 @@ func (c *Client) delete(metadata unversioned.ResourceMetadata, helper conversion
 		return err
 	}
 
+	// Convert the Metadata to a Key and combine with the Metadata revision to create
+	// a KVPair for the delete operation.  At the moment only the WorkloadEndpoint Get
+	// operations fills in the revision information.
 	if k, err := helper.convertMetadataToKey(metadata); err != nil {
 		return err
-	} else if err := c.Backend.Delete(&model.KVPair{Key: k}); err != nil {
+	} else if err := c.Backend.Delete(&model.KVPair{Key: k, Revision: metadata.GetObjectMetadata().Revision}); err != nil {
```
Comment (gunjan5) [inline lib/client/client.go]: does this (`metadata.GetObjectMetadata().Revision` ) panic if `Revision` is `nil`?

Comment (robbrockbank) [inline lib/client/client.go]: Nope.  The Revision has always been present in the backend.
